contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/insertion/InsertionDetourTimeCalculator.java

```diff
@@ -157,8 +154,33 @@ private PickupTimeInfo calculatePickupIfSameLink(VehicleEntry vEntry, int pickup
 			// case 3: previous waypoint is an ongoing (started) stop
 			// insertion is the beginning of the planned stop (traditionally)
 			
-			// IRTX TODO: This should actually become the *actual* time of insertion which
-			// may be *now*, i.e. a couple of seconds after the task has been started
+			/*
+			 * TODO: Generally, DRT is allowed to insert pickups into ongoing stop tasks.
+			 * However, currently, stop tasks have a fixed duration that is never changed.
+			 * This means that a pickup is added to an ongoing task that will end, maybe in
+			 * 20s. The task will then end regardless of the added pickup. The request will,
+			 * hence, have a wait time of 20s although the configured stop duration may be
+			 * 60s. We can even frequently have requests with zero wait time.
+			 * 
+			 * To mitigate the problem, the following steps are necessary:
+			 * - Impose here not the stopTask.beginTime as the time at which the request is inserted,
+			 *   but impose the current time ("now") as the point of insertion.
```

Comment:
We could provide both: (1) the start of the task and (2) the current time. This way we can also get the old behaviour.

I am not saying that the old behaviour is best. To be honest, I would like to have a minimum viable flexibility, so the optimizer code is as simple as possible. I could also imagine that at some point it's easier to have two separate insertion algorithms, each supporting a subset of features than to have one super-flexible that is hard to understand, maintain and fix.

Reply:
We pass the DrtStopTask, which contains the stop begin time. So we can exactly replicate the existing behavior, which is also the stanard implementation. So in the standard setting, this is only another parameter that is pasesd to the StopTimeCalculator, but not used. However, for more complex implementation it can be used. In general the idea was to touch the insertion code as little as possible in this PR and extract the stop time logic into the StopTimeCalculator.

Reply:
I understood your comment that you want to change the current behaviour. Now I got it. Thanks.